vnext/Microsoft.ReactNative/Fabric/Composition/Modal/WindowsModalHostViewComponentView.cpp

```diff
@@ -137,7 +193,6 @@ LRESULT CALLBACK ModalBoxWndProc(HWND hwnd, UINT message, WPARAM wparam, LPARAM
       return 0;
     }
     case WM_DESTROY: { // called when we want to destroy the window
-      data->Release();
       SetProp(hwnd, CompHostProperty, nullptr);
```

Comment:
Check this is getting hit.  -- right now I dont think the window is being destroyed.

Reply:
Fixed! Though this did point me to the case of closing the Modal with the "exit" window button, currently not implemented but will create a follow-up issue for it